### ISSUE TITLE
Register user plugins for specific filetypes

### ISSUE THREAD

**Issue by reegnz** [rmagatti/gx-extended.nvim#14]
Besides regex, the original gx-extended plugin allowed for an additional filetype-based filtering, so that the regex matching was only happening in a specific filetype. The filetype detection itself is then handled by vim instead of the plugin.
Only workaround today is defining your own filename pattern, but then you start reimplementing an inferior version of the ftdetect functionality which is already part of vim.

**Comment by rmagatti** [comment]
Right. I can see it being useful to have a `ft` based implementation as well. It would bifurcate the interface of how to register handlers but that might be okay

**Comment by reegnz** [comment]
@rmagatti I opened a PR with an implementation, I hope that works out.
